### ISSUE TITLE
Questions about training.

### ISSUE THREAD

**Issue by wngh1187** [BUTSpeechFIT/EEND#6]
Hi. I am getting a lot of help from your amazing research. 
Thanks again for making your code public.

By the way, I am now facing difficulties in reproducing the EEND-EDA system.
When training with ns2_beta2 data, I get a dev DER of 28.62% at 100 epochs.

This is much different from the performance reported in the original paper (2.69%), could you please share the pre-trained model weights?

In my experiment, the training standard loss and training attractor loss are reported as 0.404 / 0.106 at 100 epochs. 
Are these loss values different from your experimental results?

**Comment by msh9184** [comment]
Hi, I have the same question too.
In my case, I obtained a DER of 11.48 at 100 epochs. (I used the [dscore repository](https://github.com/nryant/dscore) for the DER calculation)

There are my [result](https://drive.google.com/file/d/1ybdb0oIc3IJt2K_DIBOa57FDCN737v-D/view?usp=sharing) and the [checkpoint](https://drive.google.com/file/d/1e6QbKpylVLj0yrY51JcPKjFmNVYycPBs/view?usp=sharing) at 100 epochs.
I trained the EEND-EDA model using a single NVIDIA RTX 3090 gpu on the default settings.

Could there be something I've missed?

**Comment by fnlandini** [comment]
Hi,
Thanks for the kind words and glad it helps.

That is indeed too far from what I would expect. I have not run too many experiments with the simulated mixtures that you are using. I only ran 5 of them to create the "SM-P" blue bar in Figure 1 here https://arxiv.org/pdf/2204.00890.pdf to show as baseline for our approach for generating simulated conversations.
Out of those 5 trainings, I only evaluated swb_sre_cv_ns2_beta2_500 (from Hitachi) for one of them and obtained 3.88 DER. Even if I score with collar 0, I get 9.56 DER so it is not that. Unfortunately, I don't have those data anymore and regenerating it is quite some work so I cannot evaluate other models.

However, in terms of training loss, all 5 runs get to the 0.1 to 0.12 ballpark for the BCE ("standard") loss and to 1e-7 to 1e-6 ballpark for the attractor loss as in the picture. So I think all of them would have similar results on that dataset.
Are you using the exact same parameters for training and inference as we [shared](https://github.com/BUTSpeechFIT/EEND/tree/main/examples)?
![image](https://github.com/BUTSpeechFIT/EEND/assets/7138393/9acc8e80-f743-4337-a323-71842124820f)

The attractor loss usually gets to virtually 0 so the differences might be related to that?

**Comment by wngh1187** [comment]
hello. Thank you for your quick reply.

I had done a batch of 600 for a quick result check.
However, it seems that this was an ineffective choice.

I retrained with the same parameters and was able to get pretty similar results to the trend in the loss graph you showed.  However, I still got a result of 18.21% DER at collar 0.
In my experiment, I noticed that the standard loss increases for 60-130 steps and then converges again. 
Is this trend normal?

![newplot](https://github.com/BUTSpeechFIT/EEND/assets/41296272/7eb0e7b9-b00c-4806-8758-7c5f9f0394e6)
![newplot (1)](https://github.com/BUTSpeechFIT/EEND/assets/41296272/0c7e9410-7e23-4b5e-9df9-ffc0ce89ed15)
![newplot (2)](https://github.com/BUTSpeechFIT/EEND/assets/41296272/ce13b968-f539-4d5b-ad43-410c0071b269)

**Comment by fnlandini** [comment]
I don't see that increase in any of my 5 trainings with these data.
Does your learning rate look like this?
![image](https://github.com/BUTSpeechFIT/EEND/assets/7138393/1fb3ad4f-0281-4835-b80a-4a2c1e27ba7d)

It makes sense that when you used a much larger batch you had different results. In this case, the learning rate scheduler reaches the maximum after 100k steps (with batchsize 64) and if you use larger batches, the same amount of steps corresponds to a much later stage in the training. In this learning rate curve you can see that the maximum LR (a bit below 2e-4) is reached after 100k steps which is a bit less than one third of the training.
I also tried batchsize 32 with 200k steps in other experiments and the performance I obtained for the final model was similar.

**Comment by wngh1187** [comment]
The running rate graph in my experiment is the same.
Maybe there's an error somewhere that I haven't found yet.
I'll check the whole process again.
Once again, thank you very much for your kind response.

**Comment by fnlandini** [comment]
Closing due to inactivity. Feel free to reopen if you see fit
